### PR TITLE
Use loglevel to dynamically change logrus' log level during runtime (#752)

### PR DESCRIPTION
**Pull request by g-linville**
for #752

This adds a little server to the controller and the api-server that listens on a local socket for requests to change the log level, and sets the logrus log level accordingly.

This also adds the `loglevel` binary to `/usr/local/bin` in the container image, which can be used like `loglevel --set info` to set the log level, or just `loglevel` to print the current log level.

This also bumps the baaah dependency in order to get the new changes that reduce the log severity on some of the common logs, as also described in #752.

### Checklist
- [x] The title of this PR would make a good line in Acorn's Release Note's Changelog
- [x] The title of this PR ends with a link to the main issue being address in parentheses, like: `This is a title (#1216)`. [Here's an example](https://github.com/acorn-io/runtime/pull/1199)
- [x] All relevant issues are referenced in the PR description. *NOTE: don't use [GitHub keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) that auto-close issues*
- [x] Commits follow [contributing guidance](https://github.com/acorn-io/runtime/blob/main/CONTRIBUTING.md#commits)
- [ ] Automated tests added to cover the changes. If tests couldn't be added, an explanation is provided in the Verification and Testing section
- [x] Changes to user-facing functionality, API, CLI, and upgrade impacts are clearly called out in PR description
- [ ] PR has at least two approvals before merging (or a reasonable exception, like it's just a docs change)

### CODE REVIEW

**Review by iwilltry42 — APPROVED** [review]

File: pkg/cli/apiserver.go [hunk 1]
```diff
@@ -40,6 +42,9 @@ func (a *APIServer) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	logrus.SetLevel(logrus.InfoLevel)
```
Comment (thedadams) [inline pkg/cli/apiserver.go:45]: Can this be added to the `StartServerWithDefaults()`?

**Comment by g-linville** [comment]
Putting this into draft until acorn-io/loglevel#1 is merged, since this is now dependent on that since we are switching to an abstract namespace unix socket.

**Review by tylerslaton — APPROVED** [review]
Looks great! Can you help me understand how the log level gets changed? I see that we have the code infrastructure in place to set it and am looking for how I trigger that outside of the endpoint.

**Comment by g-linville** [comment]
> Looks great! Can you help me understand how the log level gets changed? I see that we have the code infrastructure in place to set it and am looking for how I trigger that outside of the endpoint.

@tylerslaton the only way to change it is to exec into the pod and run the `loglevel` command. From the issue description:

> First, provide a way to dynamically set the log level of the acorn controller or api server. As an example, rancher did this via: https://github.com/rancher/loglevel/tree/master and https://github.com/rancher/log/blob/master/server/server.go. This allows an operator to run a command against the pod that will change log level with out redploying

**Review by thedadams — APPROVED** [review]

**Pull request merged**